canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/WebrtcStorageCanaryConsumer.java

```diff
@@ -0,0 +1,215 @@
+package com.amazon.kinesis.video.canary.consumer;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.ArrayList;
+import java.util.concurrent.Future;
+import java.lang.Exception;
+import java.text.MessageFormat;
+
+import com.amazonaws.auth.EnvironmentVariableCredentialsProvider;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatch;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchClientBuilder;
+import com.amazonaws.services.kinesisvideo.AmazonKinesisVideo;
+import com.amazonaws.services.kinesisvideo.AmazonKinesisVideoClientBuilder;
+import com.amazonaws.services.kinesisvideo.model.APIName;
+import com.amazonaws.services.kinesisvideo.model.GetDataEndpointRequest;
+import com.amazonaws.services.kinesisvideo.model.TimestampRange;
+import com.amazonaws.services.kinesisvideo.model.FragmentSelector;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+import com.amazonaws.services.cloudwatch.model.MetricDatum;
+import com.amazonaws.services.cloudwatch.model.PutMetricDataRequest;
+import com.amazonaws.services.cloudwatch.model.StandardUnit;
+import com.amazonaws.services.kinesisvideo.model.StartSelector;
+import com.amazonaws.services.kinesisvideo.model.StartSelectorType;
+import com.amazonaws.kinesisvideo.parser.utilities.FrameVisitor;
+import com.amazonaws.kinesisvideo.parser.examples.GetMediaWorker;
+
+import lombok.extern.log4j.Log4j2;
+
+/*
+ * Canary for WebRTC with Storage thro Media Server
+ * 
+ * For longRun-configured jobs, this Canary will emit FragmentContinuity metrics by continuously
+ * checking for any newly ingested fragments for the given stream. The fragment list is checked
+ * for new fragments every "fragmentDuration * 2" The fragmentDuration is set by the encoder in
+ * configuration in Media Server.
+ * 
+ * For periodic-configured jobs, this Canary will emit TimeToFirstFragment metrics by continuously
+ * checking for consumable media from the specified stream via GetMedia calls.
+ */
+
+@Log4j2
+public class WebrtcStorageCanaryConsumer {
+    protected static Date canaryStartTime;
+    protected static String streamName;
+    private static String canaryLabel;
+    private static String region;
+    private static EnvironmentVariableCredentialsProvider credentialsProvider;
+    private static AmazonKinesisVideo amazonKinesisVideo;
+    private static AmazonCloudWatch cwClient;
+
+    private static void calculateFragmentContinuityMetric(CanaryFragmentList fragmentList) {
+        try {
+            final GetDataEndpointRequest dataEndpointRequest = new GetDataEndpointRequest()
+                    .withAPIName(APIName.LIST_FRAGMENTS).withStreamName(streamName);
```

Comment:
If they are not static, the following build error is emitted: `non-static method calculateFragmentContinuityMetric() cannot be referenced from a static context`. I think they need to be static because although main() is a part of the WebrtcStorageCanaryConsumer class, main() is a static function, and there is no instance instantiated of the WebrtcStorageCanaryConsumer class.